device/iot-device-samples/send-receive-x509-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/SSLContextBuilder.java

```diff
@@ -0,0 +1,276 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package samples.com.microsoft.azure.sdk.iot;
+
+import org.apache.commons.codec.binary.Base64;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.UUID;
+
+/**
+ * Helper class that demonstrates how to build an SSLContext for x509 authentication from your public and private certificates,
+ * or how to build an SSLContext for SAS authentication from the default IoT Hub public certificates
+ */
+public class SSLContextBuilder
```

Comment:
This same class exists in another sample already. Copying it over to this sample as well since it is needed here as well

Reply:
This sample used to use a device client constructor that read certificates from strings, but that was one of the constructors to go. Now users need to pass in the SSLContext directly if they want to parse certificates

Reply:
In .NET we'd case this as `SslContextBuilder`. Not the same for Java?